From b1b91eff075c687f7e3cf85ab30082e74efb3abe Mon Sep 17 00:00:00 2001
From: TS-ANC <208963744+TS-ANC@users.noreply.github.com>
Date: Tue, 9 Dec 2025 15:24:00 +0100
Subject: [PATCH 3/5] Enable libcxx assertions

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8cadd8bf..9f9940d58 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,6 +65,9 @@ else()
   set(CMAKE_SHARED_LINKER_FLAGS_TRITONRELBUILDWITHASSERTS "/debug:fastlink /INCREMENTAL")
 endif()
 
+#add_definitions(-U_GLIBCXX_ASSERTIONS)
+#add_definitions(-U_LIBCPP_ENABLE_ASSERTIONS)
+
 # Default build type
 if(NOT CMAKE_BUILD_TYPE)
   message(STATUS "Default build type: Release")
-- 
2.39.5 (Apple Git-154)

