From 0c8b2725a78a081859b9890b7475644ed3f6adf4 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 7 May 2023 14:53:42 +1000
Subject: [PATCH 5/8] Unvendor third party libs

---
 CMakeLists.txt  |  3 ++-
 python/setup.py | 11 ++++++-----
 2 files changed, 8 insertions(+), 6 deletions(-)

Index: triton/CMakeLists.txt
===================================================================
--- triton.orig/CMakeLists.txt	2023-11-03 14:18:30.069609247 -0500
+++ triton/CMakeLists.txt	2023-11-03 14:18:33.778883924 -0500
@@ -45,7 +45,8 @@
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 # Third-party
-include_directories(${PYBIND11_INCLUDE_DIR})
+# include_directories(${PYBIND11_INCLUDE_DIR})
+find_package(pybind11 REQUIRED)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -D__STDC_FORMAT_MACROS  -fPIC -std=gnu++17 -fvisibility=hidden -fvisibility-inlines-hidden")
 
Index: triton/python/setup.py
===================================================================
--- triton.orig/python/setup.py	2023-11-03 14:16:34.619103000 -0500
+++ triton/python/setup.py	2023-11-03 14:18:52.648016415 -0500
@@ -90,7 +90,7 @@
 
 
 def get_thirdparty_packages(triton_cache_path):
-    packages = [get_pybind11_package_info(), get_llvm_package_info()]
+    packages = [get_llvm_package_info()]
     thirdparty_cmake_args = []
     for p in packages:
         package_root_dir = os.path.join(triton_cache_path, p.package)
@@ -225,8 +225,9 @@
             "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON",
             "-DPYTHON_INCLUDE_DIRS=" + python_include_dir,
         ]
-        if lit_dir is not None:
-            cmake_args.append("-DLLVM_EXTERNAL_LIT=" + lit_dir)
+        # cmake_args.extend()
+        # if lit_dir is not None:
+        #     cmake_args.append("-DLLVM_EXTERNAL_LIT=" + lit_dir)
         cmake_args.extend(thirdparty_cmake_args)
 
         # configuration
