From 90f2db11d694cd158fa4418ff60d2688c023e025 Mon Sep 17 00:00:00 2001
From: TS-ANC <208963744+TS-ANC@users.noreply.github.com>
Date: Tue, 9 Dec 2025 16:03:57 +0100
Subject: Enable PATH fallback for finding CUDA tools

---
 third_party/nvidia/backend/compiler.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/third_party/nvidia/backend/compiler.py b/third_party/nvidia/backend/compiler.py
index 6db76a352..335b0e68d 100644
--- a/third_party/nvidia/backend/compiler.py
+++ b/third_party/nvidia/backend/compiler.py
@@ -11,6 +11,7 @@ import re
 import tempfile
 import signal
 import os
+import shutil
 import subprocess
 from pathlib import Path
 import sysconfig
@@ -36,6 +37,7 @@ def _path_to_binary(binary: str):
     paths = [
         os.environ.get(f"TRITON_{binary.upper()}_PATH", ""),
         os.path.join(os.path.dirname(__file__), "bin", binary),
+        shutil.which(binary),
     ]
 
     for path in paths:
-- 
2.39.5 (Apple Git-154)

