From e50454a16bafa90bcd028b2eb6a6a173bd6e15db Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 7 May 2023 14:55:38 +1000
Subject: [PATCH 8/8] Search for libs in CONDA_PREFIX instead of third_party
 directory

Index: triton/python/triton/common/build.py
===================================================================
--- triton.orig/python/triton/common/build.py	2024-04-01 14:40:04.013678079 -0500
+++ triton/python/triton/common/build.py	2024-04-02 15:18:31.053369980 -0500
@@ -59,7 +59,7 @@
 @functools.lru_cache()
 def cuda_include_dir():
     base_dir = os.path.join(os.path.dirname(__file__), os.path.pardir)
-    cuda_path = os.path.join(base_dir, "third_party", "cuda")
+    cuda_path = os.path.join(os.environ['CONDA_PREFIX'])
     return os.path.join(cuda_path, "include")
 
 
