Index: triton/python/setup.py
===================================================================
--- triton.orig/python/setup.py	2023-11-06 15:55:07.943992000 -0600
+++ triton/python/setup.py	2023-11-06 15:56:34.359990824 -0600
@@ -77,7 +77,7 @@
     elif system == "Linux":
         vglibc = tuple(map(int, platform.libc_ver()[1].split('.')))
         vglibc = vglibc[0] * 100 + vglibc[1]
-        linux_suffix = 'ubuntu-18.04' if vglibc > 217 else 'centos-7'
+        linux_suffix = 'ubuntu-18.04'
         system_suffix = f"linux-gnu-{linux_suffix}"
     else:
         return Package("llvm", "LLVM-C.lib", "", "LLVM_INCLUDE_DIRS", "LLVM_LIBRARY_DIR", "LLVM_SYSPATH")
