diff --git a/CMakeLists.txt b/CMakeLists.txt
index de6ed2393..0a63e52ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,6 +65,9 @@ else()
   set(CMAKE_SHARED_LINKER_FLAGS_TRITONRELBUILDWITHASSERTS "/debug:fastlink /INCREMENTAL")
 endif()
 
+#add_definitions(-U_GLIBCXX_ASSERTIONS)
+#add_definitions(-U_LIBCPP_ENABLE_ASSERTIONS)
+
 # Default build type
 if(NOT CMAKE_BUILD_TYPE)
   message(STATUS "Default build type: Release")
@@ -143,11 +146,22 @@ endfunction()
 
 # Disable warnings that show up in external code (gtest;pybind11)
 if(NOT MSVC)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-covered-switch-default -fvisibility=hidden")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-covered-switch-default -fvisibility=hidden")
 else()
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4244 /wd4624 /wd4715 /wd4530")
 endif()
 
+# use system mlir-tblgen if provided
+if(DEFINED ENV{MLIR_TBLGEN_EXE})
+  set(MLIR_TABLEGEN_EXE $ENV{MLIR_TBLGEN_EXE})
+  message(STATUS "Using system MLIR tablegen: ${MLIR_TABLEGEN_EXE}")
+endif()
+
+if(DEFINED MLIR_TBLGEN_EXE)
+  set(MLIR_TABLEGEN_EXE ${MLIR_TBLGEN_EXE})
+  message(STATUS "Using system MLIR tablegen: ${MLIR_TABLEGEN_EXE}")
+endif()
+
 include_directories(".")
 include_directories(${MLIR_INCLUDE_DIRS})
 include_directories(${LLVM_INCLUDE_DIRS})
diff --git a/python/setup.py b/python/setup.py
index 065b170aa..ac32a5b6d 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -451,6 +451,10 @@ class CMakeBuild(build_ext):
             cmake_args.append("-DLLVM_EXTERNAL_LIT=" + lit_dir)
         cmake_args.extend(thirdparty_cmake_args)
 
+        # use system mlir-tblgen
+        cmake_args.append(f"-DMLIR_TBLGEN_EXE={os.environ.get('MLIR_TBLGEN_EXE', 'mlir-tblgen')}")
+        cmake_args.append(f"-DMLIR_TABLEGEN_EXE={os.environ.get('MLIR_TABLEGEN_EXE', 'mlir-tblgen')}")
+
         # configuration
         cfg = get_build_type()
         build_args = ["--config", cfg]
diff --git a/python/triton/runtime/build.py b/python/triton/runtime/build.py
index 1b76548d4..e59c9c303 100644
--- a/python/triton/runtime/build.py
+++ b/python/triton/runtime/build.py
@@ -1,5 +1,7 @@
+import sys
 import sysconfig
 import os
+import platform
 import shutil
 import subprocess
 
@@ -28,6 +30,28 @@ def _build(name, src, srcdir, library_dirs, include_dirs, libraries):
     py_include_dir = sysconfig.get_paths(scheme=scheme)["include"]
     custom_backend_dirs = set(os.getenv(var) for var in ('TRITON_CUDACRT_PATH', 'TRITON_CUDART_PATH'))
     include_dirs = include_dirs + [srcdir, py_include_dir, *custom_backend_dirs]
+
+    #include_dirs = [srcdir, py_include_dir, *custom_backend_dirs]
+
+    # add conda-forge specific include directories
+    # See packages cuda-cudart-dev on each related platform for the location
+    # of the cuda.h header
+    if platform.system() == "Linux":
+        arch = platform.machine()
+
+        arch_to_directory = {
+            "ppc64le": "targets/ppc64le-linux/include",
+            "aarch64": "targets/sbsa-linux/include",
+            "x86_64": "targets/x86_64-linux/include",
+        }
+
+        arch_dependent_include = arch_to_directory.get(arch, None)
+
+    if arch_dependent_include is not None:
+        include_dirs.append(os.path.join(sys.prefix, arch_dependent_include))
+    elif platform.system() == "Windows":
+        include_dirs.append(os.path.join(sys.prefix, "Library", "include"))
+
     # for -Wno-psabi, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=111047
     cc_cmd = [cc, src, "-O3", "-shared", "-fPIC", "-Wno-psabi", "-o", so]
     cc_cmd += [f'-l{lib}' for lib in libraries]
diff --git a/third_party/nvidia/backend/compiler.py b/third_party/nvidia/backend/compiler.py
index 6db76a352..335b0e68d 100644
--- a/third_party/nvidia/backend/compiler.py
+++ b/third_party/nvidia/backend/compiler.py
@@ -11,6 +11,7 @@ import re
 import tempfile
 import signal
 import os
+import shutil
 import subprocess
 from pathlib import Path
 import sysconfig
@@ -36,6 +37,7 @@ def _path_to_binary(binary: str):
     paths = [
         os.environ.get(f"TRITON_{binary.upper()}_PATH", ""),
         os.path.join(os.path.dirname(__file__), "bin", binary),
+        shutil.which(binary),
     ]
 
     for path in paths:
